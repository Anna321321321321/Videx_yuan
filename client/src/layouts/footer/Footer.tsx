import { Icon, Layout, Modal } from 'antd';
import moment from 'moment';
import React from 'react';
import * as Logger from 'videx/client/logger';
import About from '../../components/about';
import OnlineFeedback from '../../components/online-feedback';

export default () => (
  <Layout.Footer
    style={{
      transition: 'opacity 0.5s',
      background: '#fff',
      marginTop: '24px'
    }}
  >
    <span>
      Â© {moment().format('YYYY')} University of British Columbia and Microsoft
      Corporation
    </span>
    <span className="videx-home-footer">
      <span>
        <OnlineFeedback />
      </span>
      <span>
        <a href="mailto:videx@ece.ubc.ca">Support</a>
      </span>
      <span>
        <a href="/terms">Terms of Use</a>
      </span>
      <span>
        <a href="/consent">Consent</a>
      </span>
      <span>
        <a href="/faq">FAQ</a>
      </span>
      <span>
        <a
          onClick={() => {
            Logger.event('Footer.About.Click');
            Modal.info({
              title: 'About ViDeX',
              content: <About />,
              okText: 'Close'
            });
          }}
        >
          About
        </a>
      </span>
    </span>
  </Layout.Footer>
);
