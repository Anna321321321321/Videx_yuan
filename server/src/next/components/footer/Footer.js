import { Layout } from 'antd';
import React, { Component } from 'react';

export default class Footer extends Component {
  render() {
    return (
      <Layout.Footer
        style={{
          transition: 'opacity 0.5s',
          background: '#fff',
          marginTop: '24px'
        }}
      >
        <div>
          Â© 2018 University of British Columbia and Microsoft Corporation
        </div>
        <div>
          <a href="mailto:videx@ece.ubc.ca">Support</a>
        </div>
        <div>
          <a href="/consent">Consent Form</a>
        </div>
        <div>
          <a href="/terms">Terms of Use</a>
        </div>
        <div>
          <a href="/faq">FAQ</a>
        </div>
      </Layout.Footer>
    );
  }
}
