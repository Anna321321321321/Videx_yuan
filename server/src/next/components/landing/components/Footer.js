import React, { Component } from 'react';
import LandingPageButtons from '../../landing-page-buttons';

export default class Footer extends Component {
  render() {
    return (
      <div className="row">
        <h1>Invited to a ViDeX course?</h1>
        <p style={{ fontSize: '14px' }}>
          You can create a profile by signing in with your Microsoft account and
          register for a course using the course registration token. ViDeX can
          only be accessed by students using a course registration token. If you
          do not have a registration token, please ask your instructor to
          provide you with one.
        </p>
        <LandingPageButtons />
      </div>
    );
  }
}
