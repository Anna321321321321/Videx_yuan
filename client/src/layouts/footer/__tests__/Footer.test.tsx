import { mount } from 'enzyme';
import React from 'react';
import Footer from '..';

describe('Footer Unit Tests', () => {
  it('Render footer', () => {
    const wrapper = mount(<Footer />);
    expect(wrapper).toMatchSnapshot();
  });
});
