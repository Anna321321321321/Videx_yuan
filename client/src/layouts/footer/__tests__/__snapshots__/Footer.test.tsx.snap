// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Unit Tests Render footer 1`] = `
<Component>
  <Adapter
    style={
      Object {
        "background": "#fff",
        "marginTop": "24px",
        "transition": "opacity 0.5s",
      }
    }
  >
    <Basic
      prefixCls="ant-layout-footer"
      style={
        Object {
          "background": "#fff",
          "marginTop": "24px",
          "transition": "opacity 0.5s",
        }
      }
    >
      <div
        className="ant-layout-footer"
        style={
          Object {
            "background": "#fff",
            "marginTop": "24px",
            "transition": "opacity 0.5s",
          }
        }
      >
        <span>
          Â© 
          2020
           University of British Columbia and Microsoft Corporation
        </span>
        <span
          className="videx-home-footer"
        >
          <span>
            <withState(Component)>
              <Component
                changeVisible={[Function]}
                visible={false}
              >
                <Icon
                  as="i"
                  color="yellow"
                  name="smile"
                >
                  <i
                    aria-hidden="true"
                    className="yellow smile icon"
                    onClick={[Function]}
                  />
                </Icon>
                <a
                  onClick={[Function]}
                >
                  Online Feedback
                </a>
                <withState(withState(withHandlers(Component)))
                  onCancel={[Function]}
                  visible={false}
                >
                  <withState(withHandlers(Component))
                    fields={
                      Object {
                        "category": Object {
                          "value": null,
                        },
                        "email": Object {
                          "value": null,
                        },
                        "message": Object {
                          "value": null,
                        },
                        "rate": Object {
                          "value": null,
                        },
                      }
                    }
                    onCancel={[Function]}
                    setFields={[Function]}
                    visible={false}
                  >
                    <withHandlers(Component)
                      fields={
                        Object {
                          "category": Object {
                            "value": null,
                          },
                          "email": Object {
                            "value": null,
                          },
                          "message": Object {
                            "value": null,
                          },
                          "rate": Object {
                            "value": null,
                          },
                        }
                      }
                      onCancel={[Function]}
                      setFields={[Function]}
                      setStatus={[Function]}
                      status={null}
                      visible={false}
                    >
                      <Component
                        fields={
                          Object {
                            "category": Object {
                              "value": null,
                            },
                            "email": Object {
                              "value": null,
                            },
                            "message": Object {
                              "value": null,
                            },
                            "rate": Object {
                              "value": null,
                            },
                          }
                        }
                        onCancel={[Function]}
                        onChange={[Function]}
                        onReset={[Function]}
                        onSubmit={[Function]}
                        setFields={[Function]}
                        setStatus={[Function]}
                        status={null}
                        visible={false}
                      >
                        <Modal
                          cancelButtonDisabled={false}
                          closable={true}
                          confirmLoading={false}
                          footer={null}
                          maskTransitionName="fade"
                          okButtonDisabled={false}
                          okType="primary"
                          onCancel={[Function]}
                          prefixCls="ant-modal"
                          title="Online Feedback"
                          transitionName="zoom"
                          visible={false}
                          width={520}
                        >
                          <DialogWrap
                            cancelButtonDisabled={false}
                            closable={true}
                            closeIcon={
                              <span
                                className="ant-modal-close-x"
                              >
                                <Icon
                                  className="ant-modal-close-icon"
                                  type="close"
                                />
                              </span>
                            }
                            confirmLoading={false}
                            footer={null}
                            maskTransitionName="fade"
                            okButtonDisabled={false}
                            okType="primary"
                            onCancel={[Function]}
                            onClose={[Function]}
                            prefixCls="ant-modal"
                            title="Online Feedback"
                            transitionName="zoom"
                            visible={false}
                            width={520}
                            wrapClassName=""
                          />
                        </Modal>
                      </Component>
                    </withHandlers(Component)>
                  </withState(withHandlers(Component))>
                </withState(withState(withHandlers(Component)))>
              </Component>
            </withState(Component)>
          </span>
          <span>
            <a
              href="mailto:videx@ece.ubc.ca"
            >
              Support
            </a>
          </span>
          <span>
            <a
              href="/terms"
            >
              Terms of Use
            </a>
          </span>
          <span>
            <a
              href="/consent"
            >
              Consent
            </a>
          </span>
          <span>
            <a
              href="/faq"
            >
              FAQ
            </a>
          </span>
          <span>
            <a
              onClick={[Function]}
            >
              About
            </a>
          </span>
        </span>
      </div>
    </Basic>
  </Adapter>
</Component>
`;
